pdns/lua-record.cc

```diff
@@ -219,6 +232,15 @@ class IsUpOracle
       state->first = false;
     }
   }
+  
+  void setWeight(const CheckDesc& cd, int weight){
+    auto statuses = d_statuses.write_lock();
+    auto& state = (*statuses)[cd];
+    state->weight = weight;
+    if (state->first) {
+      state->first = false;
+    }
```

Comment:
Since this duplicates work done in `setStatus` (and thus in `setUp` and `setDown`), and `setWeight` is always followed by `setUp`, there is no need to keep these lines.
```suggestion
```

Or you might consider keeping that and adding `state->status = true`, and remove the `setUp` call following the `setWeight` call, since a non-zero weight implies being up.

Reply:
updated the setweight to only sets weight and setUp will updates the `first` check flag